From: John Johansen <john.johansen@canonical.com>
Date: Fri, 13 Mar 2026 01:58:03 -0700
Subject: [PATCH 1/5] parser: add more reserved mediation classes

Update the parser for reserved mediation classes that are comming.

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 parser/policydb.h |  9 ++++++++-
 parser/rule.cc    | 12 ++++++------
 2 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/parser/policydb.h b/parser/policydb.h
index 5606c4e..422a014 100644
--- a/parser/policydb.h
+++ b/parser/policydb.h
@@ -44,7 +44,14 @@
 #define AA_CLASS_DISPLAY_LSM	20
 #define AA_CLASS_NS		21
 #define AA_CLASS_IO_URING	22
-
+/* 23 unallocated */
+#define AA_CLASS_PGID		24
+#define AA_CLASS_SID		25
+#define AA_CLASS_SCHED		26
+#define AA_CLASS_IOPRI		27
+#define AA_CLASS_PRCTL		28
+/* 29 unallocated */
+#define AA_CLASS_NETV9_SKB	30
 #define AA_CLASS_X		31
 /* defined in libapparmor's apparmor.h #define AA_CLASS_DBUS 32 */
 
diff --git a/parser/rule.cc b/parser/rule.cc
index e4fff0b..db89447 100644
--- a/parser/rule.cc
+++ b/parser/rule.cc
@@ -44,13 +44,13 @@ const char *aa_class_table[] = {
 	"userns",
 	"io_uring",
 	"unknown23",
-	"unknown24",
-	"unknown25",
-	"unknown26",
-	"unknown27",
-	"unknown28",
+	"pgid",
+	"sid",
+	"scheduler",
+	"iopriority",
+	"prctl",
 	"unknown29",
-	"unknown30",
+	"netv9_skb",
 	"X",
 	"dbus",
 	NULL
